cli/tests/e2e/test_match_based_id.py

```diff
@@ -30,36 +30,24 @@ def test_duplicate_matches_indexing(run_semgrep_in_tmp, snapshot):
         ("rules/match_based_id/join.yaml", "join.py", True),
     ],
 )
-def test_id_change(run_semgrep_in_tmp, snapshot, rule, target, change):
-    # Since the id is based on the file name, we want the exact same file
-    # everytime, for both the before and after files
-    with tempfile.NamedTemporaryFile(dir=Path("targets")) as tf:
-        with open(Path("targets/match_based_id/before") / target) as fin:
-            tf.write(fin.read().encode("utf-8"))
-        tf.flush()  # Make sure file has been copied.
-        tf.seek(
-            0
-        )  # Seek to beginning since Semgrep will be reading from it. Just in case.
-        before_results, _errors = run_semgrep_in_tmp(
-            rule,
-            target_name=tf.name,
-            output_format=OutputFormat.JSON,
-        )
-        before_id = json.loads(before_results)["results"][0]["extra"]["fingerprint"]
-        tf.seek(0)  # Seek to beginning again so we can read after file
-        tf.truncate()  # delete before content
-        tf.flush()  # Make sure file has been copied.
-        with open(Path("targets/match_based_id/after") / target) as fin:
-            tf.write(fin.read().encode("utf-8"))
-        tf.flush()  # Make sure file has been copied.
-        tf.seek(
-            0
-        )  # Seek to beginning since Semgrep will be reading from it. Just in case.
-        after_results, _errors = run_semgrep_in_tmp(
+def test_id_change(run_semgrep_on_copied_files, tmp_path, rule, target, expect_change):
+
+    suffix = Path(target).suffix
+
+    # Static file path (necessary to obtain static id)
+    static_target = tmp_path / "targets" / ("_match_based_id" + suffix)
```

Comment:
I don't entirely follow how this aligns with the description I'm seeing for the `target_name` parameter

>     :param target_name: which directory within ./e2e/targets/ to scan

but I'm not that familiar with the Python codebase.

Reply:
Hm I'm not sure why the description says which directory, it should be which directory or file

Reply:
Fair; that description is probably incomplete; the path must be _within_ `<cwd>/e2e/targets`, but does not necessarily need to be specified _relative to_ `<cwd>/e2e/targets`.

Note that in this case, the current working directory is `tmp_path`; I specified the target as an absolute path so that things will bomb out if, for whatever reason, the working directory changes.

What's the best way to document this so it's clear?

Reply:
Maybe something like

:param target_name: which file or directory to scan. This must either be a relative path or an absolute path of a file contained in ./e2e/targets

If I understand you correctly?

Reply:
sgtm; I updated this and other documentation

Reply:
lgtm then